06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js

```diff
@@ -0,0 +1,141 @@
+const Index = require('../Model/inverted-index.js');
+const test = require('../files/test.json');
+const work = require('../files/work.json');
+const empty = require('../files/empty.json');
+
+const fileAPI = require('file-api');
+
+const File = fileAPI.File;
+
+
+const invertedIndex = new Index();
+
+const workBook = [{
+  title: 'Checkpoint one',
+  text: 'Testing getIndex function',
+},
+
+{
+  title: 'Checkpoint Two',
+  text: 'Coming soon',
+},
+];
+describe('InvertedIndex : define methods', () => {
+  invertedIndex.createIndex('work.json', workBook);
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.getIndex).toBeDefined();
+    expect(invertedIndex.createIndex).toBeDefined();
+    expect(invertedIndex.tokenize).toBeDefined();
+    expect(Index.validateFile).toBeDefined();
+    expect(invertedIndex.searchIndex).toBeDefined();
+    expect(invertedIndex.flattenSearch).toBeDefined();
+  });
+});
+
+describe('InvertedIndex : createIndex', () => {
+  invertedIndex.createIndex('work.json', workBook);
+  it('should verify that index has been created', () => {
+    expect(Object.keys(invertedIndex.getIndex('work.json', workBook)).length)
+      .toBeGreaterThan(0);
+  });
+  it('Should return an object that is an accurate index of the file uploaded',
+    () => {
+      expect(invertedIndex.getIndex('work.json', workBook)).toEqual({
+        checkpoint: [0, 1],
+        one: [0],
+        testing: [0],
+        getindex: [0],
+        function: [0],
+        two: [1],
+        coming: [1],
+        soon: [1],
+      });
+    });
+  it('should verify that index has been created', () => {
+    invertedIndex.createIndex('test.json', test);
+    expect(Object.keys(invertedIndex.getIndex('test.json', test)).length)
+      .toBeGreaterThan(0);
+  });
+});
+
+describe('InvertedIndex : searchIndex', () => {
+  it('Should return correct index of the search term in work.json', () => {
+    expect(invertedIndex.searchIndex('soon', 'work.json')).toEqual({
+      soon: [1],
+    });
+  });
+  it('should return true if the constant is a string', () => {
+    const words = 'Alice in Wonderland';
+    expect(Object.keys(invertedIndex.searchIndex(words))).toBeTruthy();
+  });
+  it('Should return correct index of the search term', () => {
+    expect(invertedIndex.searchIndex('humans', 'test.json')).toEqual({
+      humans: [1],
+    });
+  });
+  it('should return tokenized words for string in variable words', () => {
+    let words = 'It CAN only BE++ $$$$$   GOD';
+    const termsExpected = 'it can only be god';
+    words = invertedIndex.tokenize(words);
+    expect(termsExpected).toEqual(words);
+  });
```

Comment:
Should this it block be part of your searchIndex() Test Suite or tokenize() Test Suite?